spinalcordtoolbox/utils/shell.py

```diff
@@ -340,16 +330,18 @@ def _split_lines(self, text, width):
         wrapped = []
         for i, li in enumerate(lines):
             if len(li) > 0:
+                # Check for ANSI graphics control sequences, and increase width to compensate
+                width_adjusted = width + len("".join(re.findall("\\x1b\[[0-9;]+m", li)))
```

Comment:
Hmm. I noticed some odd text-wrapping behavior:

![image](https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/16181459/62144259-e422-42cc-8afa-9abeae94083a)

Interestingly, the `-----` divider text doesn't get wrapped (80 characters wide). However, the "DESCRIPTION" section seems to get wrapped early, even when there is space left.

When I open up a Python console and print the `list_tasks_string` directly, everything fits:

![image](https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/16181459/43d8b99e-03c0-42d7-863c-d70d74763c88)

I think this is happening because of the [ANSI graphics control sequences](https://notes.burke.libbey.me/ansi-escape-codes) we add to the string to give it color (e.g. `\x1b[91m`). So, I think we just need to account for the space taken up by these codes when wrapping.

I've added some patches to `shell.py` that do this. And, the wrapping is now performed as expected (before/after):

![Screenshot from 2023-09-11 14-12-52](https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/16181459/ce4d0f56-9278-422b-81f9-cb2dfef3b0a2)

I just need to make sure that my patches do not affect any other argparse help descriptions across our other CLI scripts.

Reply:
I did a very silly comparison, [diffing the entire text of the Command-Line Tools page](https://www.diffchecker.com/Jnd10Eca/) before and after my patch.

It looks like the only thing affected by the patch is the `sct_deepseg` table. :slightly_smiling_face: